frontend/src/models/propertyDefinitionsModel.ts

```diff
@@ -94,7 +117,8 @@ export const propertyDefinitionsModel = kea<
         ],
         propertyDefinitions: [
             (s) => [s.propertyStorage],
-            (propertyStorage): PropertyDefinition[] => propertyStorage.results || [],
+            (propertyStorage): PropertyDefinition[] =>
+                [...reservedProperties, ...propertyStorage.results] || [...reservedProperties],
```

Comment:
`[...reservedProperties, ...propertyStorage.results]` always results in at least an empty array, so the `|| [...reservedProperties]` part is never called. You can just omit it.